File: src/api/src/main/java/fr/iut/uca/model/issues/IssueModelInfo.java
package fr.iut.uca.model.issues;

public class IssueModelInfo {

    protected String name;

    protected String shortDescription;

    protected String description;

    public IssueModelInfo(String name, String shortDescription, String description) {
        this.name = name;
        this.shortDescription = shortDescription;
        this.description = description;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getShortDescription() {
        return shortDescription;
    }

    public void setShortDescription(String shortDescription) {
        this.shortDescription = shortDescription;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }
}

File: src/app/model/issues/IssueModelInfo.ts
import {Category} from "./Category";
import {IssueModelShortInfo} from "./IssueModelShort";

export class IssueModelInfo extends IssueModelShortInfo{
  private readonly _id: string
  private _category: Category

  constructor(id: string, name: string, shortDescription: string, description: string, category: Category) {
    super(name, shortDescription, description)
    this._id = id;
    this._category = category;
  }

  get id(): string {
    return this._id;
  }

  get category(): Category {
    return this._category;
  }

  set category(value: Category) {
    this._category = value;
  }
}
